### PR TITLE
Validate attrs in Resolver constructor

### PR DESCRIPTION
**Pull request by tmquinn**
Unblocks the default tests. Test framework requires Resolver to be createable without a namespace.
Deleted route/application-test as there is no application route

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ Ember.Registry.prototype.normalize = function(i) { return i; }
 
 export default class Resolver {
   constructor(attrs) {
-    this.namespace = attrs.namespace;
-    this.modulePrefix = this.namespace.modulePrefix;
+    if (attrs) {
+      this.namespace = attrs.namespace;
+    }
```
Comment (stefanpenner) [inline addon/index.js]: if there is no namespace, wont `moduleNameForFullName` fail?

File: package.json [hunk 2]
```diff
@@ -21,26 +21,26 @@
...
   },
   "dependencies": {
-    "ember-cli-babel": "^6.1.0"
+    "ember-cli-babel": "^6.0.0"
   },
   "devDependencies": {
     "broccoli-asset-rev": "^2.4.5",
-    "ember-ajax": "^2.4.1",
-    "ember-cli": "~2.13.0-beta.3",
+    "ember-ajax": "^3.0.0",
```
Comment (stefanpenner) [inline package.json]: i think we can actually delete this entry

File: tests/dummy/config/environment.js [hunk 3]
```diff
@@ -1,10 +1,9 @@
 /* eslint-env node */
-'use strict';
 
 module.exports = function(environment) {
-  let ENV = {
+  var ENV = {
     modulePrefix: 'dummy',
-    environment,
+    environment: environment,
```
Comment (stefanpenner) [inline tests/dummy/config/environment.js]: It seems strange to change this to ES5, but then also depend on ES6 features (like in tests/helpers/module-for-acceptance.js)

File: tests/helpers/module-for-acceptance.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ export default function(name, options = {}) {
 
     afterEach() {
       let afterEach = options.afterEach && options.afterEach.apply(this, arguments);
-      return resolve(afterEach).then(() => destroyApp(this.application));
+      return Promise.resolve(afterEach).then(() => destroyApp(this.application));
```
Comment (stefanpenner) [inline tests/helpers/module-for-acceptance.js]: 👍

File: tests/dummy/config/environment.js (hunk 3 above)
Comment (tmquinn) [inline tests/dummy/config/environment.js]: yah, this happened as part of the `ember init` phase of upgrading ember-cli. It was one of those things where I figured someone must have changed it for a reason...

File: addon/index.js (hunk 1 above)
Comment (tmquinn) [inline addon/index.js]: The tests harness uses test-helpers/resolver.js. For some reason, the namespaces, is set AFTER create

**Pull request merged**